torch/distributed/distributed_c10d.py

```diff
@@ -1562,6 +1562,17 @@ def _tensor_to_object(tensor, tensor_size):
     buf = tensor.numpy().tobytes()[:tensor_size]
     return _unpickler(io.BytesIO(buf)).load()
 
+def _check_for_nccl_backend(group):
+    pg = group or _get_default_group()
+    # It is not expected for PG to be wrapped many times, but support it just
+    # in case
+    while isinstance(pg, _ProcessGroupWrapper):
+        pg = pg.wrapped_pg
```

Comment:
This can cause errors because `_ProcessGroupWrapper` may be undefined (https://github.com/pytorch/pytorch/blob/e7906d57d7d7db0486b41a01de13fc2fc0baf5cf/torch/distributed/distributed_c10d.py#L56-L60) . 

@rohan-varma can you take a look?

Reply:
@ppwwyyxx Thanks for raising this! Is it causing any internal issue?